### ISSUE TITLE
Ubuntu/Linux fails on installation

### ISSUE THREAD

**Issue by reecube** [Armax/Pokemon-GO-node-api#109]
When I try to install the module with npm on Ubuntu/Linux, I get the following error:

```
...
../src/angle.h:23:27: error: ISO C++ forbids declaration of ‘parameter’ with no type [-fpermissive]
../src/angle.h:23:47: error: expected ‘,’ or ‘...’ before ‘<’ token
     static void New(const FunctionCallbackInfo<Value>& args);
                                               ^
../src/angle.h:24:33: error: ‘FunctionCallbackInfo’ does not name a type
     static void Normalize(const FunctionCallbackInfo<Value>& args);
                                 ^
../src/angle.h:24:33: error: ISO C++ forbids declaration of ‘parameter’ with no type [-fpermissive]
../src/angle.h:24:53: error: expected ‘,’ or ‘...’ before ‘<’ token
     static void Normalize(const FunctionCallbackInfo<Value>& args);
                                                     ^
make: *** [Release/obj.target/NativeExtension/src/NativeExtension.o] Error 1
make: Leaving directory `/home/***********/node-pokemon-go-api-example/node_modules/pokemon-go-node-api/node_modules/s2geometry-node/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/share/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:797:12)
gyp ERR! System Linux 3.16.0-30-generic
gyp ERR! command "nodejs" "/usr/bin/node-gyp" "rebuild"
gyp ERR! cwd /home/***********/node-pokemon-go-api-example/node_modules/pokemon-go-node-api/node_modules/s2geometry-node
gyp ERR! node -v v0.10.25
gyp ERR! node-gyp -v v0.10.10
gyp ERR! not ok
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian

npm http 304 https://registry.npmjs.org/is-buffer
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"v0.10.25","npm":"1.3.10"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"v0.10.25","npm":"1.3.10"})
npm ERR! weird error 1
npm WARN optional dep failed, continuing uglify-js@2.7.0
npm ERR! not ok code 0
```

What should I do?

**Comment by TomVivant** [comment]
I changed my server's os to Arch Linux cause of this... and it worked !

**Comment by reecube** [comment]
@TomVivant  how/where did you do this?

**Comment by bencevans** [comment]
Checkout https://github.com/Armax/Pokemon-GO-node-api/issues/60#issuecomment-234402303

**Comment by reecube** [comment]
@bencevans Thank you for the link. I've switched my npm version. Now it seems to work, but I think using an old npm and node version is not a real solution. Especially has the npm version 2 some critical packaging bugs that are resolved on version 3.

**Comment by bencevans** [comment]
@reecube agreed, have you isolated it to check if the npm version is the issue though, or is it just the version of Node?

**Comment by reecube** [comment]
@bencevans how can I isolate it?

**Comment by bencevans** [comment]
``` bash
##
# node@4.x npm@2.x
##
# Use Node.js 4.x
$ nvm use 4

# Check versions
$ node -v
v4.4.7

$ npm -v
2.15.8

# Check Installs
$ npm install
 ...
$ echo $?
0 # 0 means successful (ok exit code)

##
# node@4.x npm@latest
##
# Upgrade npm to latest
$ npm install -g npm@latest
npm@3.10.5

# Remove node_modules and attempt another install
$ rm -rf node_modules
$ npm install
...
$ echo $?
0 # successful

##
# node@6.x npm@latest
##
$ nvm use 6
$ npm install --global npm@latest
$ rm -rf node_modules
$ npm install
...
$ echo $?
0 # successful
```

I think it's been fixed now, I've just pulled the latest master and got it installing on node@6 and node@4 and varied npm versions. Try a `git pull` and `npm install` again. :smile_cat:

**Comment by bencevans** [comment]
89be0c955f730fadcc1028dac1b2c2d177a2113f may well have fixed it by bumping the version of the dependency that was failing the install.

**Comment by reecube** [comment]
@bencevans very nice, thank you a lot!

**Comment by coolaj86** [comment]
Fixed in #165 by using pure JavaScript - no C++ or compiler necessary.
